docs/DeveloperGuide.md

```diff
@@ -1,34 +1,920 @@
+<br/>
+
+Given below is the example usage scenario and how the `Bookmark` command behaves at each step.
+
+> :bulb: The command is using one-based indexing while the program stores and recognises the zero-based indexing. The following example will use one-based to be consistent with the command. 
+
+**Step 1:** User inputs command `bookmark`,  the application then calls `Parser#getCommand()` and passes the command to it.
+
+**Step 2:** `BookmarkParser` extracts “Bookmark Action” from the command, and based on the command type, it will determine the respective required field. The `BookmarkParser` is responsible for validating the input parameters and will throw an `AniException` when encountering unrecognised inputs.
+
+For example: `bookmark 2 -e 5`, the parser will create `BookmarkCommand` and call the method setBookmarkAction(“e”), setBookmarkIndex(2) and setBookmarkEpisode(5).
+
+**Step 3:** The application calls `BookmarkCommand#execute()` and the command will use the `BookmarkAction` to do a corresponding validation check on the field before calling the  Bookmark Operations.
+
+Below is a list of bookmark operations:
+*   `Bookmark#addAnimeBookmark()`: Adds the `Anime` index provided into the bookmark list.
+*   `Bookmark#getListInString()`: List all entries within the `Bookmark` using the `Bookmark` index together with the `Anime` name.
+*   `Bookmark#deleteAnimeBookmark()`: Remove the `Bookmark` index provided from the bookmark list.
+*   `Bookmark#editBookmarkEpisode()`: Edit the current episode for a Bookmark entry .
+*   `Bookmark#addNote()`: Add a note for a Bookmark entry.
+*   `Bookmark#removeNote()`: Remove a note from a Bookmark entry.
+*   `Bookmark#getAnimeInfoFromBookmark()`: Retrieve the anime info on the anime id kept at the specified bookmark entry.
+*   `Bookmark#getBookmarkEpisode()`: Retrieve the episode field for a Bookmark entry.
+*   `Bookmark#getNoteInString()`: Retrieve notes in a list within the specified bookmark entry.
+
+**Step 4:** The user executes `bookmark -a 430` command to add the anime id: 3 into the bookmark. `Bookmark#addAnimeBookmark()` will then add the anime index to the ArrayList within the bookmark.
+
+![Bookmark State After Add Diagram](images/Bookmark-After-Step4.png) <br/>
+*Figure 25: Bookmark Entry After Add*
+
+> :memo: The table shows the three ArrayList objects in the column with the bookmark id. When adding a new anime id into the bookmark, the bookmark will initialise the anime episode to be 0 together with an empty note object.
+
+**Step 4.5:** The user executes `bookmark -a 1` and `bookmark -a 410` to add anime id 1 and 410 to the bookmark.
+
+![Bookmark State After More Add Diagram](images/Bookmark-After-Step4.5.png) <br/>
+*Figure 26: Bookmark Entries with more Add*
+
+The following sequence diagram shows how the `Add Bookmark` operation works:
+
+![Bookmark Add Command Sequence Diagram](images/Bookmark-Add-Sequence-Diagram.png) <br/>
+*Figure 27: Bookmark Add Command Sequence Diagram*
```

Comment:
![image](https://user-images.githubusercontent.com/57039314/97527233-c64be580-19e5-11eb-99ab-25a129d61d9c.png)

Good job on the sequence diagram, everything looks to be in-check! Easy to understand 👍